lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -293,6 +293,49 @@ static EnumDecl *validateCodingKeysType(const DerivedConformance &derived,
   return codingKeysDecl;
 }
 
+/// WARNING: Copied from CSSimplify.cpp
+///
+/// Produce a score (smaller is better) comparing a parameter name and
+/// potentially-typo'd argument name.
+///
+/// \param paramName The name of the parameter.
+/// \param argName The name of the argument.
+/// \param maxScore The maximum score permitted by this comparison, or
+/// 0 if there is no limit.
+///
+/// \returns the score, if it is good enough to even consider this a match.
+/// Otherwise, an empty optional.
+///
+static Optional<unsigned> scoreParamAndArgNameTypo(StringRef paramName,
```

Comment:
Let's lift this utility into the `TypeChecker` namespace instead of duplicating it here.

Reply:
In fact, if it's possible to generalize the surrounding code in CSSimplify to suit your needs here that would be even cooler, but totally not necessary.